grant-project/zcash-grant-system
# Proposal arbiter

Part of #31.

Proposal arbiter basics. Milestone arbitration & payout flow is next.

### What this does
- adds `Proposal`:`User` (many:1) arbiter relationship
- admin: filter by "Arbiter: missing"
- admin: home 'proposals needing arbiters' action items
- admin: set/change arbiter for proposal
- email: notify the user that has been assigned arbiter
- email notification setting for arbiter related things
- frontend: user profile "arbitrations" tab

### Notes
- Profile - Tabs are getting crowded
- the arbitrations tab is just a list of user's arbiter proposals, will get more goodies with milestone stuff

### Testing
1. upgrade database
1. make sure you have a LIVE (published) proposal
1. open admin home page, should be an action item for proposals needing arbiters, click the link
1. you should see the proposals list page filtered by LIVE & ARBITER
1. select one of the proposals, you should see an alert in the detail with a set arbiter button
1. click the set arbiter button and select an arbiter who's email you can check
1. check that you received the arbiter email
1. check the arbiter user's "arbitrations" profile tab and make sure there is an entry for the correct proposal

## Screenies
<img width="984" alt="screen shot 2019-02-06 at 4 59 15 pm" src="https://user-images.githubusercontent.com/11430954/52380144-4bcd7c00-2a32-11e9-997b-27b0fc5b075f.png">
<img width="865" alt="screen shot 2019-02-06 at 4 59 34 pm" src="https://user-images.githubusercontent.com/11430954/52380152-51c35d00-2a32-11e9-889f-670ef859e669.png">
<img width="772" alt="screen shot 2019-02-06 at 5 02 19 pm" src="https://user-images.githubusercontent.com/11430954/52380160-55ef7a80-2a32-11e9-98eb-d3a573e6b316.png">
<img width="866" alt="screen shot 2019-02-06 at 5 03 24 pm" src="https://user-images.githubusercontent.com/11430954/52380167-5ab42e80-2a32-11e9-97a6-9b6e14e196db.png">
